### PR TITLE
Imgur API support

### PR DESCRIPTION
**Pull request by pajlada**
xd

### CODE REVIEW

File: internal/resolvers/imgur/load.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	"github.com/Chatterino/api/pkg/resolver"
+)
+
+func buildTooltip(miniData miniImage) (interface{}, error, time.Duration) {
+	var tooltip bytes.Buffer
+
+	if err := imageTooltipTemplate.Execute(&tooltip, &miniData); err != nil {
+		return &resolver.Response{
+			Status:  http.StatusInternalServerError,
+			Message: "Tweet template error: " + resolver.CleanResponse(err.Error()),
```
Comment (leon-richardt) [inline internal/resolvers/imgur/load.go:19]: This is not a Tweet :smiley:

File: internal/resolvers/imgur/resolver.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+	"log"
+
+	"github.com/Chatterino/api/pkg/cache"
+	"github.com/Chatterino/api/pkg/resolver"
+	"github.com/koffeinsource/go-imgur"
+)
+
+var (
+	timestampFormat = "Jan 2 2006 • 15:04 UTC"
```
Comment (leon-richardt) [inline internal/resolvers/imgur/resolver.go:16]: Not necessarily for this PR but maybe we should consider storing a common timestamp format for all resolvers that use one

**Review by leon-richardt — CHANGES_REQUESTED** [review]

File: internal/resolvers/imgur/resolver.go (hunk 2 above)
Comment (pajlada) [reply internal/resolvers/imgur/resolver.go:16]: Agreed :+1:

Comment (pajlada) [reply internal/resolvers/imgur/resolver.go:16]: https://github.com/Chatterino/api/issues/82

File: internal/resolvers/imgur/check.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+import (
+	"net/url"
+	"strings"
+)
+
+func check(url *url.URL) bool {
+	// TODO: Make a shared helper function that does basically this.
+	// helpers.IsSubdomainOf(url.Host, "imgur.com")
+	isImgur := strings.HasSuffix(url.Host, ".imgur.com") || url.Host == "imgur.com"
```
Comment (zneix) [inline internal/resolvers/imgur/check.go:11]: Are we just straight up accepting a wildcard for `*.imgur.com`? How about domains like `https://api.imgur.com/` should we also support them?

**Review by pajlada — COMMENTED** [review]
LGTM :+1: 
image

File: internal/resolvers/imgur/check.go (hunk 3 above)
Comment (pajlada) [reply internal/resolvers/imgur/check.go:11]: This is handled in imgur/load.go:40 where we return the `ErrDontHandle` error to signify that the generic resolver should step in whenever a non-image URL has been handled

**Pull request merged**